src/PoolManager.sol

```diff
@@ -49,6 +50,9 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
 
     constructor(uint256 controllerGasLimit) Fees(controllerGasLimit) {}
 
+    // todo change to transient storage
+    address currentHook;
```

Comment:
maybe it makes sense to remove this until we have the transient keyword - given youre not using it

Reply:
or instead actually i think you could remove the library, and wherever you want to write

```
currentHook = hook
```

just write
```
assembly { tstore(currentHook.slot, hook) }
```

and similarly you can use
```
address hook;
asembly {hook := tload(currentHook.slot)}
```

Reply:
could put those in helper functions for now if you want to but theyre fairly small